### PR TITLE
WiP: Add OpenStack support

### PR DESCRIPTION
**Pull request by dulek**
This is very strongly WiP, but adding it here for teammates reference.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/kas/params.go [hunk 1]
```diff
@@ -335,6 +336,9 @@ func NewKubeAPIServerParams(ctx context.Context, hcp *hyperv1.HostedControlPlane
 	case hyperv1.AzurePlatform:
 		params.CloudProvider = azure.Provider
 		params.CloudProviderConfig = &corev1.LocalObjectReference{Name: manifests.AzureProviderConfigWithCredentials("").Name}
+	case hyperv1.OpenStackPlatform:
+		params.CloudProvider = openstack.Provider
+		// TODO(dulek): Cloud provider config here?
```
Comment (dulek) [inline control-plane-operator/controllers/hostedcontrolplane/kas/params.go:341]: KAS has to have CloudProviderConfig!

**Comment by dulek** [comment]
/remove-lifecycle stale

I'm going to keep this opened, there's a rising interest in actually going beyond PoC here.

**Comment by EmilienM** [comment]
I'm going to try taking over the PR ownership since Michal doesn't work with us anymore, it'll help us to debug directly in the CI jobs.
/close

**Comment by EmilienM** [comment]
We now have a PR owned by myself: https://github.com/openshift/hypershift/pull/4289

**Pull request closed without merging**